Agents/PubChemAgent/README.md

```diff
@@ -0,0 +1,195 @@
+# (can be left blank for Stack deployment)
+- QUERY_ENDPOINT=
+- UPDATE_ENDPOINT=
+```
+
+The STACK_NAME variable is used to identify the deployment mode of the agent. In case the STACK_NAME is left blank, default Blazegraph endpoint setting will be taken from the docker-compose file. Otherwise they will be retrieved using the StackClients based on the provided NAMESPACE variable.
+
+Please note:
+
+All variables defined here (except for STACK_NAME) serve as default values. To omit any of those default values, either remove the key completely or just leave it blank.
+
+A missing STACK_NAME variable will result in an error; however, when deploying using the stack-manager start up script, the STACK_NAME variable will be set automatically for all services. Hence, this could be left blank here; however, if provided, it needs to match the STACK_NAME used by the stack-manager!
+
+### Build and publish the Docker image
+
+To build and publish the agent Docker image please use the following commands. Please note that all of those commands are bundled in the publish_docker_image.sh convenience script (only target image (i.e. production/debug) needs to be adjusted at top of that script).
+
+```bash
+# Building the Docker image (production / debug)
+docker-compose -f docker-compose.yml  build
+docker-compose -f docker-compose.debug.yml  build
+
+# Publish the Docker image to the Github container registry
+docker image push ghcr.io/cambridge-cares/<image tag>:<version>
+```
+
+Time out issues have been observed when building the image. If this happens, please try pulling the required stack-clients image first by
+
+```bash
+docker pull docker.cmclinnovations.com/stack-client:1.6.2
+```
+
+### Docker deployment
+
+Deploy the dockerised agent by running the following code in the command prompt from the same location where this README is located (ideally, use a bash terminal to avoid potential issues with inconsistent path separators).
+
+```bash
+# Deploy the Docker images (production / debug) locally
+docker-compose -f docker-compose.yml  up
+docker-compose -f docker-compose.debug.yml  up
+```
+
+To verify the correct startup of the agent, open the URL address the agent is running on, e.g. <http://localhost:5000> in your browser.
+
+### Stack Deployment
+
+If you want to spin up this agent as part of a stack, do the following:
+
+- Build the production image using the commands provided above (do not spin up the image)
+- Copy the pubchem-agent.json file from the stack-manager-input-config folder into the inputs/config/services folder of the stack manager
+- Copy os_stack.json into the inputs/config folder of the stack manager
+- Start the stack manager (i.e. bash ./stack.sh start os_stack from the stack-manager repo). This should start the container. Please use a bash terminal to avoid potential issues with inconsistent path separators.
+- The agent shall become available at ```<http://<HOST>:<PORT>/>```
```

Comment:
You may want to replace the `<HOST>` and `<PORT>` placeholders with something useful, and put the complete address, e.g. using the default stack you'd get `http://localhost:3838/pubchemagent/`.

Reply:
changed as follow:

- Start the stack manager running the following command from the stack-manager repo:

bash ./stack.sh start os_stack <PORT>

Please use a bash terminal to avoid potential issues with inconsistent path separators.
The port is optional and defaults to 3838. This should start the container. The agent shall become available at <http://localhost:PORT/pubchemagent/> (e.g. <http://localhost:3838/pubchemagent/> if using the default port).